docs/upgrading.md

```diff
@@ -0,0 +1,73 @@
+# Upgrading the Agent
+
+This document provides user guidance around agent upgrades and versioning.
+Like every other piece of software included in your project, the agent 
+should be upgraded frequently enough to pick up ongoing enhancements 
+while allowing room for testing. 
+
+## Watching For New Releases
+
+How can you find out about new releases? The best way is to "Watch" this repo
+from the "Watch" button in the upper-right corner of the main repository 
+page. If you don't want to be notified for every pull request and issue, you can 
+choose "Custom" under "Watch" and select just the "Releases" checkbox.
+
+Another approach is to just check in regularly with the GitHub 
+[releases page](https://github.com/signalfx/splunk-otel-java/releases) for this 
+project. 
+
+## When Does Splunk Release?
+
+Because we are a downstream distribution of [OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation),
+we generally release within a few working days of every upstream release. We also release when we 
+have new features or enhancements that we want to make available.
+
+We contribute to upstream OpenTelemetry and are typically aware of the release cycle
+over there. As a result, depending on the timing of certain patch releases, we may sometimes
```

Comment:
As a result of what? This sentence doesn't follow, because the previous sentences suggested that you don't skip any releases. 

Please explain "release thrash" or state exactly what you mean. 

I think that, in this section, you're trying to indicate the following:

1. Because Observability Cloud OpenTelemetry is downstream of OpenTelemetry Java Distribution, Observability Cloud releases within a few days of an OpenTelemetry Java Distribution, with rare exceptions.
2. Observability Cloud releases Observability Cloud OpenTelemetry to provide new Observability Cloud OpenTelemetry features.
3. Observability Cloud sometimes skips a release of OpenTelemetry Java Distribution, so that releases don't occur too often for our customers to consume them.

Reply:
> As a result of what?

As a result of us being deeply involved in the upstream development and staying abreast of the upstream release cycle.

Ugh, I think all this "observability" and "cloud" stuff just confuses the situation and is hard for me to read! 🙃 

1. Yes, we release are distro within a few days of an upstream release, with rare exceptions.
2. Yes, releases are made to provide new features/enhancements/fixes.
3. Yeah, you've got the gist of what I meant by "thrash". If there are 3 upstream patch releases in 3 days, it doesn't make sense for our distro to match those 1-to-1, and we'd probably roll them into a single patch release, especially because we're in the loop and find out about stuff quickly.